modules/core/src/main/scala/org/tesselation/modules/Services.scala

```diff
@@ -20,24 +20,28 @@ object Services {
     cfg: AppConfig,
     nodeId: PeerId,
     keyPair: KeyPair,
-    storages: Storages[F]
+    storages: Storages[F],
+    queues: Queues[F]
   ): F[Services[F]] =
     for {
       _ <- Async[F].unit
       healthcheck = HealthCheck.make[F]
       session = Session.make[F](storages.session, storages.cluster, storages.node)
       cluster = Cluster
         .make[F](cfg, nodeId, keyPair, storages.session)
+      gossip = Gossip.make[F](queues.rumor, keyPair)
```

Comment:
Imo `gossip <- Gossip.make[F](keyPair)` which already encapsulates creating of `Queue` inside and "bounds" it to the Gossip instance by returning `F[Gossip[F]]`. See my previous comment :)

Reply:
Saw it and I disagree with it.